experimental/Schemes/Schemes.jl

```diff
@@ -0,0 +1,2587 @@
+# homomorphisms.
+#
+@Markdown.doc """
+    function compose( f::AffSchMorphism, g::AffSchMorphism ) 
+
+Returns the composition f ∘ g of two morphisms of affine schemes. 
+NOTE: The ordering of the arguments follows the mathematical convention, 
+unlike other functions for composition in Oscar! 
+"""
+function compose( f::AffSchMorphism, g::AffSchMorphism ) 
+  if codomain(g) != domain(f) 
+    error( "morphisms can not be composed" )
+  end
+  # TODO: Extend functionality to also work for the fractional 
+  # representations of morphisms.
+  phi = pullback(f)
+  gamma = pullback(g)
+  return AffSchMorphism( domain(g), codomain(f), compose( phi, gamma ) )
+end
+
+@Markdown.doc """
+    mutable struct Glueing
+
+Glueing of two affine schemes X and Y along principal open subsets 
+U ⊂ X and V ⊂ Y via an isomorphism Φ : U → V. 
+A glueing consists of a diagram X ↩ U → V ↪ Y with the outer maps 
+the open embeddings of principal open subsets. 
+"""
+mutable struct Glueing
+  firstPatch::AffineScheme
+  secondPatch::AffineScheme
+  inclusionFirstPatch::AffSchMorphism
+  inclusionSecondPatch::AffSchMorphism
+  glueingIsomorphism::AffSchMorphism
+
+  function Glueing( 
+      inclusionFirstPatch, 
+      inclusionSecondPatch, 
+      glueingIsomorphism 
+    )
+    if domain( inclusionFirstPatch ) != domain( glueingIsomorphism ) 
+      error( "Morphisms can not be composed for glueing" )
+    end
+    if codomain( glueingIsomorphism ) != domain( inclusionSecondPatch ) 
+      error( "Morphisms can not be composed for glueing" )
+    end
+    G = new()
+    G.firstPatch = codomain( inclusionFirstPatch )
+    G.secondPatch = codomain( inclusionSecondPatch )
+    G.inclusionFirstPatch = inclusionFirstPatch
+    G.inclusionSecondPatch = inclusionSecondPatch
+    G.glueingIsomorphism = glueingIsomorphism
+    return G
+  end
+end
+
+# Glueing for two principal open subsets of a common root scheme X
+# along their intersection in X
+@Markdown.doc """
+    function Glueing( U::SpecPrincipalOpen, V::SpecPrincipalOpen )
+
+Provides the glueing between two principal open subsets U and V of 
+some common parent X in their localization trees. If no such common 
+parent is found, this necessarily returns an error. Otherwise, the 
+intersection U ∩ V ⊂ X is computed and the glueing returned. 
+"""
+function Glueing( U::SpecPrincipalOpen, V::SpecPrincipalOpen )
+  root(U) == root(V) || error( "schemes do not belong to the same root" )
+  X = root(U)
+  R = ambient_ring(X)
+  den_U = denoms(U)
+  den_V = denoms(V)
+  f = prod(den_U)
+  g = prod(den_V)
+  U_loc = localize(U,g)
+  V_loc = localize(V,f)
+  iso = AffSchMorphism( U_loc, V_loc, gens(R) )
+  return Glueing( inclusion_in_parent(U_loc), inclusion_in_parent(V_loc), iso )
+end
+
+@Markdown.doc """
+    first_patch( G::Glueing ) = G.firstPatch
+
+Returns the first patch for the glueing G.
+"""
+first_patch( G::Glueing ) = G.firstPatch
+@Markdown.doc """
+    second_patch( G::Glueing ) = G.secondPatch
```

Comment:
```suggestion
    second_patch( G::Glueing )
```
No, the body of a one-line function still does not belong into the doc comment, please also fix this in other places